pkg/service/replicator/replicate.go

```diff
@@ -0,0 +1,220 @@
+package replicator
```

Comment:
Similar to above, I would move the private replicate function to `pkg/service/storage/handlers/replicate/replicate.go`
- define an interface for the things it needs on the storage service:
```
ID()
Receipts()
Blobs()
PDP()
Claims()
UploadService()
```
Signature for Replicate becomes Replicate(ctx,service, task)

- Move all the job queue machinery inside the storage service or move it a much smaller service for just the jobqueue

Reply:
mostly focused on seperate of concerns between stateless replication handling and job scheduling

Reply:
ooo yeah, I like that much better. Done in latest change.